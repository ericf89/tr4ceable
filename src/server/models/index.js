import Router from 'express-promise-router';
import userRoutes from './user/routes';
import authRoutes from './auth/routes';

const router = Router();

router.use('/auth', authRoutes);
router.use('/users', userRoutes);

export default router;

