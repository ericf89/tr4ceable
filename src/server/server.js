import express from 'express';
import bodyParser from 'body-parser';
import config from 'config';
import compression from 'compression';
import morgan from 'morgan';
import winston from 'winston';

export const server = () => {
  const app = express();
  const development = config.get('nodeEnv') === 'development';

  app.locals.pretty = development;
  app.set('port', config.get('port'));
  app.disable('x-powered-by');
  app.use(bodyParser.json());
  app.use(compression());
  app.use(morgan(development ? 'dev' : 'combined'));

  app.get('*', (req, res) => res.status(200).json({}));
  return app;
};

export default () => server().listen(config.get('port'), () => winston.info('Listening on port %d', config.get('port')));

