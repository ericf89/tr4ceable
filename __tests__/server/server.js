import supertest from 'supertest';
import { server } from '../../src/server/server';

describe('basic server', () => {
  it('should return empty json', async () => {
    const resp = await supertest(server()).get('/');
    expect(resp.status).toBe(200);
    expect(resp.text).toMatchSnapshot();
  });
});
