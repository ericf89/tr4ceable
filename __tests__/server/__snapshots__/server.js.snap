// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic server should return empty json 1`] = `"{}"`;
